### PR TITLE
0532_K-diff_Pairs_in_an_Array.py

### LINKED ISSUE #263: 0532 - K-diff Pairs in an Array
## Description of the Problem
Given an array of integers nums and an integer k, return the number of unique k-diff pairs in the array.

A k-diff pair is an integer pair (nums[i], nums[j]), where the following are true:

+ 0 <= i, j < nums.length
+ i != j
+ a <= b
+ b - a == k

## Code
```
from collections import Counter
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        ans = 0
        c = Counter(nums)
        if k == 0:
            for val in c.values():
                if val > 1:
                    ans += 1
            return ans
        for i in c.keys():
            if i + k in c.keys():
                ans += 1
        return ans
```

## Link To The LeetCode Problem
[LeetCode](https://leetcode.com/problems/k-diff-pairs-in-an-array/)

### LINKED ISSUE #273: 1288 - Remove Covered Intervals
## Description of the Problem
Given a list of intervals, remove all intervals that are covered by another interval in the list.

Interval [a,b) is covered by interval [c,d) if and only if c <= a and b <= d.

After doing so, return the number of remaining intervals.

## Code
```
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: (x[0], -x[1]))
        inside = 0
        right = -1
        for i, j in intervals:
            if j <= right:
                inside += 1
            else:
                right = j
        return len(intervals) - inside
```

## Link To The LeetCode Problem
[LeetCode](https://leetcode.com/problems/remove-covered-intervals/)

### PR DESCRIPTION
**Pull request by anuranjanpandey**
**IMPORTANT: Please do not create a Pull Request without an issue.**

### All Submissions:

* [x] I have read the CONTRIBUTING document.
* [x] My code is written in **Python3** and is ending with _.py_
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] I have checked that my submission does pass the test on LeetCode.com 
* [x] Does your filename follow the naming Conventions?
* [x] Have you linked your PR to an Issue?

**Closing issues**

Put `closes #263 ` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: LeetCode/0523_K_diff_Pairs_in_an_Array.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from collections import Counter
+class Solution:
+    def findPairs(self, nums: List[int], k: int) -> int:
```
Comment (vJechsmayr) [inline LeetCode/0523_K_diff_Pairs_in_an_Array.py:3]: Make sure you fit the naming of the code from the issue.

**Review by vJechsmayr — CHANGES_REQUESTED** [review]
Please check your Algorithm once more - there is an problem with an Test case on LeetCode:

image

[https://leetcode.com/problems/continuous-subarray-sum/submissions/](https://leetcode.com/problems/continuous-subarray-sum/submissions/)

File: LeetCode/0523_K_diff_Pairs_in_an_Array.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        c = Counter(nums)
+        if k == 0:
+            for val in c.values():
+                if val > 1:
+                    ans += 1
+            return ans
+        for i in c.keys():
+            if i + k in c.keys():
+                ans += 1
+        return ans
```
Comment (Vulwsztyn) [inline LeetCode/0523_K_diff_Pairs_in_an_Array.py]: no endline at the end of file

**Comment by anuranjanpandey** [comment]
> Please check your Algorithm once more - there is an problem with an Test case on LeetCode:
> 
> image
> 
> https://leetcode.com/problems/continuous-subarray-sum/submissions/

Screenshot (59)

**Review by vJechsmayr — APPROVED** [review]
Thank you for contributing to my #hacktoberfest Repository!
Happy Hacktoberfest!

**Pull request merged**